### ISSUE TITLE
ALLOWED_DOMAINS requires explicit www.

### ISSUE THREAD

**Issue by jamesoncollins** [jamesoncollins/turbo-bot#8]
There should be no difference between example.com and www.example.com if the point is that we're checking for a domain.

so, either change the logic hat checks the url, or make sure that we can have a optional wildcard for www. in ALLOWED_DOMAINS

**Comment by jamesoncollins** [comment]
possibly a better method for extract_url:

```
def contains_url_with_domain(substring, domain):
    """
    Checks if a substring contains a URL with a specific domain.

    Args:
        substring: The substring to check.
        domain: The domain name to search for.

    Returns:
        True if the substring contains a URL with the specified domain, False otherwise.
    """
    # Regular expression to match URLs with the specified domain
    url_regex = r"https?://(?:www\.)?" + re.escape(domain) + r"(/[\w\-\.\?%&=]*)?"

    # Search for the URL pattern in the substring
    match = re.search(url_regex, substring)

    return bool(match)
```

**Comment by jamesoncollins** [comment]
should be fixed here a7f3485ef387f8588b0da31edf360e2f0cf4757d
